apps/web/src/views/Home/components/CakeSection/CakeSectionTag.tsx

```diff
@@ -0,0 +1,219 @@
+import { useTranslation } from '@pancakeswap/localization'
+import {
+  EarnIcon,
+  GovernanceIcon,
+  InsertChartOutlinedIcon,
+  NftIcon,
+  PoolIcon,
+  StoreIcon,
+  SwapIcon,
+  Text,
+  TrophyIcon,
+} from '@pancakeswap/uikit'
+import useTheme from 'hooks/useTheme'
+import { cloneElement, useMemo, useRef, useLayoutEffect, useState } from 'react'
+import Marquee from 'react-fast-marquee'
+import { styled } from 'styled-components'
+
+import { PartnerApolloX, PartnerAlpaca, PartnerVenus, PartnerLedger, PartnerMetaMask } from './PartnerLogos'
+
+const MARQUEE_WIDTH = 210
+
+export const usePartnerData = () => {
+  const { t } = useTranslation()
+  return useMemo(() => {
+    return [
+      { icon: <PartnerLedger />, width: 21, text: t('Ledger') },
+      { icon: <PartnerMetaMask />, width: 20, text: t('Metamask') },
+      { icon: <PartnerVenus />, width: 20, text: t('Venus') },
+      { icon: <PartnerAlpaca />, width: 20, text: t('Alpaca Finance') },
+      { icon: <PartnerApolloX />, width: 20, text: t('ApolloX') },
+    ]
+  }, [t])
+}
+
+export const useEcosystemTagData = () => {
+  const { t } = useTranslation()
+  return useMemo(() => {
+    return [
+      { icon: <PoolIcon />, text: t('Staking') },
+      { icon: <EarnIcon />, text: t('Farming') },
+      { icon: <SwapIcon />, text: t('Trade') },
+      { icon: <NftIcon />, text: t('NFT') },
+      {
+        icon: <InsertChartOutlinedIcon />,
+        text: t('Liquidity Provision'),
+      },
+      {
+        icon: <TrophyIcon />,
+        text: t('Game'),
+      },
+      { icon: <GovernanceIcon />, text: t('Governance') },
+      { icon: <StoreIcon />, text: t('IFO') },
+    ]
+  }, [t])
+}
```

Comment:
Can we make all of this be like

const partnerData = [
 { icon: <PartnerLedger />, width: 21, text: <Trans>Ledger</Trans> },
]